### PR TITLE
Add common multi-bus routines

### PR DESCRIPTION
**Pull request by frinzell**
Adding some commonly used routines to read and write multiple instances of a register, augmenting multi-bus support.

### CODE REVIEW

File: chipsec/chipset.py [hunk 1]
```diff
@@ -839,6 +849,12 @@ def read_register(self, reg_name, cpu_thread=0, bus_index=0):
 
         return reg_value
 
+    def read_register_all(self, reg_name, cpu_thread=0):
+        values = []
+        for index in range( len(self.get_register_bus( reg_name )) ):
```
[the remark this answers is not in the document]
Comment (frinzell) [reply chipsec/chipset.py]: Register instances are accessed by their Bus index.

File: chipsec/chipset.py [hunk 2]
```diff
@@ -855,6 +865,15 @@ def read_register(self, reg_name, cpu_thread=0, bus_index=0):
 
         return reg_value
 
+    def read_register_all(self, reg_name, cpu_thread=0):
+        values = []
+        bus_data = self.get_register_bus( reg_name )
+        if bus_data is None:
+            return None
```
Comment (BrentHoltsclaw) [inline chipsec/chipset.py]: Should return the result of self.read_register( reg_name, cpu_thread)

File: chipsec/chipset.py [hunk 3]
```diff
@@ -855,6 +865,15 @@ def read_register(self, reg_name, cpu_thread=0, bus_index=0):
...
+        for index in range( len(bus_data) ):
```
Comment (BrentHoltsclaw) [inline chipsec/chipset.py]: get_register_bus returns a list of the buses, should just loop through those instances.  The instances are returned in string, so you'll need to convert them to integer.

Comment (BrentHoltsclaw) [reply chipsec/chipset.py]: @frinzell please let me know why range is being used instead of the list of bus entries

Comment (frinzell) [reply chipsec/chipset.py]: `index` in `self.read_register()`  is index (0,1,2...) of the bus list not the actual bus number (0x1D, 0x1F,..)

**Comment by BrentHoltsclaw** [comment]
@frinzell please rebase

**Pull request merged**